apps/codelab/src/app/components/slides/title-slide/title-slide.component.ts

```diff
@@ -9,4 +12,27 @@ export class TitleSlideComponent {
   @Input() title: string;
   @Input() description: string;
   @Input() prereqs: string;
+
+  constructor(
+    private readonly activeRoute: ActivatedRoute,
+    private readonly router: Router,
+    private readonly presentation: SlidesDeckComponent,
+    @Inject(MENU_ROUTES) private readonly menuRoutes
+  ) {
+    if (this.presentation != null) {
+      this.setupPrevious();
+    }
+  }
+
+  private setupPrevious() {
+    const config = this.activeRoute.snapshot.pathFromRoot
```

Comment:
We're we planning to move this logic to a shared service since we're actually just copy-pasting it accross 2 components?